manipulation/station.py

```diff
@@ -265,6 +265,195 @@ def add_directives(
+    package_xmls: typing.List[str] = [],
+    parser_preload_callback: typing.Callable[[Parser], None] = None,
+    parser_prefinalize_callback: typing.Callable[[Parser], None] = None,
+):
+    """See MakeMultibodyPlant and MakeRobotDiagram for details."""
+    ApplyMultibodyPlantConfig(scenario.plant_config, plant)
+    for p in package_xmls:
+        parser.package_map().AddPackageXml(p)
+    ConfigureParser(parser)
+    if parser_preload_callback:
+        parser_preload_callback(parser)
+
+    # Make the plant for the iiwa controller to use.
+    flattened_directives = FlattenModelDirectives(
+        ModelDirectives(directives=scenario.directives), parser.package_map()
+    ).directives
+    children_to_freeze = set()
+    if model_instance_names and add_frozen_child_instances:
+        children_to_freeze = _FindChildren(flattened_directives, model_instance_names)
+    all_model_instances = children_to_freeze.union(model_instance_names)
+
+    directives = []
+    for d in flatted_directives:
+        if d.add_model and (d.add_model.name in model_instance_names):
+            directives.append(d)
+        if (
+            d.add_weld
+            and (
+                ScopedName.Parse(d.add_weld.child).get_namespace()
+                in all_model_instances
+            )
+            and (
+                d.add_weld.parent == "world"
+                or ScopedName.Parse(d.add_weld.parent).get_namespace()
+                in all_model_instances
+            )
+        ):
+            directives.append(d)
+    ProcessModelDirectives(
+        directives=ModelDirectives(directives=directives),
+        parser=parser,
+    )
+
+    # TODO(russt): Replace all joints in children_to_freeze with weld joints.
+
+    if parser_prefinalize_callback:
+        parser_prefinalize_callback(parser)
+
+    plant.Finalize()
+
+
+def MakeMultibodyPlant(
+    scenario: Scenario,
+    *,
+    model_instance_names: typing.List[str] = None,
+    add_frozen_child_instances: bool = True,
+    package_xmls: typing.List[str] = [],
+    parser_preload_callback: typing.Callable[[Parser], None] = None,
+    parser_prefinalize_callback: typing.Callable[[Parser], None] = None,
+):
+    """Use a scenario to create a MultibodyPlant. This is intended, e.g., to facilitate
+    easily building subsets of a scenario, for instance, to make a plant for a
+    controller.
+
+    Args:
+        scenario: A Scenario structure, populated using the `load_scenario`
+            method.
+
+        model_instance_names: If specified, then only the named model instances
+            will be added to the plant. Otherwise, all model instances will be added.
+
+        add_frozen_child_instances: If True and model_instance_names is not None, then
+            model_instances that are not listed in model_instance_names, but are welded
+            to a model_instance that is listed, will be added to the plant; with all
+            joints replace by welded joints.
```

Comment:
typo: "replaced"

On that note, what kind of comments should we include in code reviews for manipulation? I'd say we want to develop fast here, so we shouldn't be picky about code style. Docstrings, type hints, and major inefficiencies are probably worth pointing out. What do you think?
It might be worth writing up a small guideline in the developer notes if we start collaborating more here (I would be happy to do this as well).